dffml/model/model.py

```diff
@@ -86,6 +86,10 @@ def __init__(self, config):
         # types.
         if isinstance(getattr(self.config, "directory", None), str):
             self.config.directory = pathlib.Path(self.config.directory)
+            # to treat "~" as the the home directory rather than a literal
```

Comment:
We still need the other `if` statement, the one in my previous comment where we check if directory is a `pathlib.Path`.

If we don't have that we'll still run into the issue this is trying to solve. Where if we run the quickstart or similar we'd find that the default parameter is being passed (which specifics home with `~`) and since it's a pathlib type and not a string, we'd end up with it not getting `.resolve()`d. So we need that if statement still

While we're at it, since we need the `directory` property for two if statements, we should probably add a local variable and assign it to `getattr(self.config, "directory", None)`.

Reply:
I have added the second if statement to make sure that the adjustment happens to pathlib.Path objects but the same test is still failing, I will try to track why this is happening

Reply:
I was able to pinpoint and fix it I think :D https://github.com/intel/dffml/pull/498